services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx

```diff
@@ -113,9 +113,15 @@ export type HealthPlanFormPageProps = {
     ) => Promise<HealthPlanPackage | Error>
 }
 export const StateSubmissionForm = (): React.ReactElement => {
-    const { id } = useParams<{ id: string }>()
+    const { id, '*': path } = useParams<{ id: string; ['*']: string }>()
+    // IF not id throw new error
+    if (!id) {
+        throw new Error(
+            'PROGRAMMING ERROR: id param not set in state submission form.'
+        )
+    }
     const { pathname } = useLocation()
-    const currentRoute = getRouteName(pathname)
+    const currentRoute = getRouteName(`/${path}`)
```

Comment:
Yeah, this is where I think we need to improve this getRouteName function. I don’t like that we are constructing a “path” thingie just to pass to that function. It’d be cleaner if we could just give the full path to it and have the function figure the right thing out from it.

Reply:
@haworku has been tackling this and made some good progress.